src/bp/ui-studio/src/web/views/OneFlow/diagram/VariablesEditor/Modal.tsx

```diff
@@ -1,40 +1,110 @@
-import { Button, FormGroup, IDialogProps, InputGroup } from '@blueprintjs/core'
-import { FlowVariable } from 'botpress/sdk'
-import { Dialog } from 'botpress/shared'
-import React, { FC, useEffect, useState } from 'react'
+import { Button, Checkbox, FormGroup, IDialogProps, InputGroup } from '@blueprintjs/core'
+import axios from 'axios'
+import { FlowVariable, FlowVariableConfig, FlowVariableParameter } from 'botpress/sdk'
+import { Dialog, Dropdown, lang, Option } from 'botpress/shared'
+import React, { FC, Fragment, useEffect, useState } from 'react'
 
 type Props = {
   variable: FlowVariable
   saveChanges: (variable: FlowVariable) => void
 } & IDialogProps
 
+const boolOptions: Option[] = [
+  { label: 'True', value: 'true' },
+  { label: 'False', value: 'false' }
+]
+
 const VariableModal: FC<Props> = props => {
+  const [variableConfigs, setVariableConfigs] = useState<FlowVariableConfig[]>()
   const [name, setName] = useState<string>()
   const [type, setType] = useState<string>()
+  const [params, setParams] = useState<any>()
+
+  const configsOptions = (variableConfigs || []).map<Option>(c => new Option(lang.tr(c.label), c.name))
+  const currentConfigOption = configsOptions.find(o => o.value === type)
+  const currentConfig = (variableConfigs || []).find(c => c.name === type)
+  const normalParams = (currentConfig?.params || []).filter(p => !p.isAdvanced)
+  const advancedParams = (currentConfig?.params || []).filter(p => p.isAdvanced)
+
+  useEffect(() => {
+    async function fetchVariables() {
+      const res = await axios.get(`${window.API_PATH}/modules/variables`)
+      setVariableConfigs(res.data)
+    }
+    fetchVariables()
+  }, [])
 
   useEffect(() => {
     setName(props.variable?.name)
     setType(props.variable?.type)
+    setParams(props.variable?.params || {})
   }, [props.variable])
 
+  const setParameter = (param: FlowVariableParameter, value) => {
+    const newParams = { ...params }
+    newParams[param.name] = value
+    setParams(newParams)
+  }
+
   const handleSaveClick = () => {
     const variable = props.variable
     variable.name = name
     variable.type = type
+    variable.params = params
     props.saveChanges(variable)
     props.onClose()
   }
 
+  const renderParameter = (param: FlowVariableParameter) => {
+    if (param.control === 'nullableCheckbox') {
+      return renderNullableCheckboxParameter(param)
+    } else {
+      return renderTextboxParameter(param)
+    }
```

Comment:
I'm wondering if this could be moddable also. So we'd have moddable ui elements for the studio